docs/source/quickstart/advanced.rst

```diff
@@ -231,3 +231,90 @@ To remove the default fields and only add the desired ones, the ``--no-default-f
 flag can be specified and then specific fields can be added in, e.g.::
 
   custodian report -s out --no-default-fields --field Image=ImageId policy.yml
+
+Server-Side Filters
+-------------------
+
+By default Cloud Custodian fetches all the resources on a cloud and filters on
+the client side. 
+In certain cases Cloud Custodian can suppport server-side filtering on certain classes of
+resources. 
+
+This might be useful if you're trying to minimize the amount of API call volume
+or response time. 
+These needs are usually specific to certain use cases and cloud resources.
+
+Caution:: Usage of this feature should consider trade offs as some features
+such as caching might not be as effective. 
+
+Support for Server-side filters is defined per-resource and handled in the
+query block of a policy.  
+
+In this example will use a server side policy to garbage collect EBS snapshots, the `query schema is defined
+here <https://github.com/cloud-custodian/cloud-custodian/blob/bb5dc8d5f1b2c9500e26d02630b64742dffcb432/c7n/resources/ebs.py#L134-L146>`_
+
+Check the documentation for `the list of
+filters <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_snapshots>`_
+, do note, they are put in the query stanza, not the filter stanza. 
+
+For example, this policy:
+
+.. code-block:: yaml
+
+  policies:
+    - name: garbage-collect--snapshots
+      resource: aws.ebs-snapshot
+      filters:
+          - type: age
+            days: 7
+            op: ge
+          - "tag:custodian_snapshot": present
+      actions:
+          - delete
+      
+If you have a large amount of snapshots, it would be inefficent to send this
+query to the client to process, so we can take advantage of some server-side
+filtering to minimize traffic. 
+
+.. code-block:: yaml
+
+  policies:
+    - name: garbage-collect-snapshots
+      resource: aws.ebs-snapshot
+      query:
+        - Name: "tag:environment"
+          Values: ["dev"]
+
+With these changes, server-side filters make sure you only query snapshots in
+the "dev" environment without needing to send the information to the client.
+
+Now let's combine server-side and client-side filtering to create a more
+efficient method of cleaning up EBS snapshots
```

Comment:
This might be out of scope, but at some point, it may be helpful to determine what we want the overall tone of these docs to be. The reason why I say this is because such a decision can then guide how we choose to address the reader. In this documentation, we address the reader as "you," which can convey a more casual tone.

Either way, establishing consistent language will be helpful. (eg, whenever we share an example, we use ''you")